### PR TITLE
[bitnami/thanos] fix: use https for queryURL when ingress is tls

### PR DESCRIPTION
**Pull request by aslafy-z**
### Description of the change

Use `https://` prefix for ruler alert URL when `queryFrontend.ingress.tls` is set to `true`. 

### Benefits

Compute a valid URL based on ingress configuration.

### Possible drawbacks

This bugfix might break some configurations in some edge cases.  

### Applicable issues

- fixes #

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by CeliaGMqrz** [comment]
Hi @aslafy-z 

Thanks for your contribution!

I have been reviewing the logic, I think we should investigate this feature as the `thanos.ruler.queryURL` function seems to be exclusively used [here](https://github.com/bitnami/charts/blob/main/bitnami/thanos/templates/ruler/statefulset.yaml#L138). However, I'm not sure, but it may be possible to implement it on [`query/deployment.yaml`](https://github.com/bitnami/charts/blob/main/bitnami/thanos/templates/query/deployment.yaml#L163) as well.

**Comment by aslafy-z** [comment]
@CeliaGMqrz I pushed an additional commit but not sure if it's an appropriate fix. What do you think of it? Should this helper be renamed? Thank you

File: bitnami/thanos/templates/query/deployment.yaml [hunk 1]
```diff
@@ -159,9 +159,7 @@ spec:
             {{- if .Values.query.grpc.client.serverName }}
             - --grpc-client-server-name={{ .Values.query.grpc.client.serverName }}
             {{- end }}
-            {{- if and .Values.queryFrontend.enabled .Values.queryFrontend.ingress.enabled .Values.queryFrontend.ingress.hostname .Values.queryFrontend.ingress.overrideAlertQueryURL }}
-            - --alert.query-url=http://{{ tpl .Values.queryFrontend.ingress.hostname . }}
-            {{- end }}
```
Comment (CeliaGMqrz) [inline bitnami/thanos/templates/query/deployment.yaml:164]: I recommend using a template. Just modify this particular line and keep the rest.
```.diff
-            - --alert.query-url=http://{{ tpl .Values.queryFrontend.ingress.hostname . }}
+            - --alert.query-url={{- template "thanos.ruler.queryURL" .}}
```

**Review by CeliaGMqrz — COMMENTED** [review]
Hi @aslafy-z,

Thanks for your feedback. I left a minor comment.

File: bitnami/thanos/templates/query/deployment.yaml (hunk 1 above)
Comment (aslafy-z) [reply bitnami/thanos/templates/query/deployment.yaml:164]: done with [`974ebf6` (#24456)](https://github.com/bitnami/charts/pull/24456/commits/974ebf6fd65d35748e7b4df9ab097d2c383fa2ad)

**Comment by aslafy-z** [comment]
@CeliaGMqrz I made the fix and rebased the PR. Can you take a look? Thank you

File: bitnami/thanos/templates/_helpers.tpl [hunk 2]
```diff
@@ -255,7 +255,11 @@ Return the queryURL used by Thanos Ruler.
 */}}
 {{- define "thanos.ruler.queryURL" -}}
 {{- if and .Values.queryFrontend.enabled .Values.queryFrontend.ingress.enabled .Values.queryFrontend.ingress.hostname .Values.queryFrontend.ingress.overrideAlertQueryURL -}}
-{{- printf "http://%s" (tpl .Values.queryFrontend.ingress.hostname .) -}}
+  {{- if .Values.queryFrontend.ingress.tls -}}
+    {{- printf "https://%s" (tpl .Values.queryFrontend.ingress.hostname .) -}}
+  {{- else -}}
+    {{- printf "http://%s" (tpl .Values.queryFrontend.ingress.hostname .) -}}
+  {{- end -}}
```
Comment (dgomezleon) [inline bitnami/thanos/templates/_helpers.tpl:262]:
```suggestion
  {{- if .Values.queryFrontend.ingress.tls -}}
    {{- printf "https://%s" (tpl .Values.queryFrontend.ingress.hostname .) -}}
  {{- else -}}
    {{- printf "http://%s" (tpl .Values.queryFrontend.ingress.hostname .) -}}
  {{- end -}}
```
```suggestion
    {{ printf "%s://%s" (ternary "https" "http" .Values.queryFrontend.ingress.tls) (tpl .Values.queryFrontend.ingress.hostname .) }}
```

**Review by dgomezleon — CHANGES_REQUESTED** [review]
Thanks for the contribution. I added a minor suggestion.

File: bitnami/thanos/templates/_helpers.tpl (hunk 2 above)
Comment (aslafy-z) [reply bitnami/thanos/templates/_helpers.tpl]: done in [`a6a9301` (#24456)](https://github.com/bitnami/charts/pull/24456/commits/a6a93016e06a351205fd0acf5c36386a31f4dab7)

**Review by dgomezleon — APPROVED** [review]
LGTM. Thanks for contributing.

**Pull request merged**